pure.zsh

```diff
@@ -324,6 +324,13 @@ prompt_pure_async_git_fetch() {
 	# Set SSH `BachMode` to disable all interactive SSH password prompting.
 	export GIT_SSH_COMMAND="${GIT_SSH_COMMAND:-"ssh"} -o BatchMode=yes"
 
+	# if gpg-agent is setup to handle ssh keys for git fetch; then
+	# make sure it uses this zpty instead of corrupting the parent tty
+	# solves issue #366
+	if export -p GPG_TTY &>/dev/null && [[ "$GPG_TTY" != "$(tty)" ]] ; then
```

Comment:
Should we check `[[ -n $GPG_TTY ]]` instead or is there special meaning if the user has done `export GPG_TTY=`? Alternatively we could do `(( $+GPG_TTY ))` which is a pretty common pattern in Zsh to check if a variable is set. We could also replace `$(tty)` with `$TTY` to avoid creating a subshell.

Or even more radical, is there a reason we shouldn't always do `export GPG_TTY=$TTY`? I can think of one: the user users a dedicated terminal for GPG auths, but I don't know if that's possible or happens in practice. But in that case the above TTY swap would also break that workflow.

Reply:
If the variable is not exported then `gpg-connect-agent` would not see it. But I accept your suggestion since the nuance isn't really important here.

I will implement your suggestions.

Reply:
done